app/__init__.py

```diff
@@ -76,6 +76,14 @@ def refresh_session():
     def markdown_filter(data):
         return Markup(markdown(data))
 
+    @application.template_filter('question_references')
+    def question_references_filter(data, get_question):
+        return re.sub(
+            r"\[\[([^\]]+)\]\]",
```

Comment:
Nice regex... maybe we could have a comment in the code to explain what it does? This feels quite obscure.